ci-operator/templates/build-azure-image-310.yaml

```diff
@@ -0,0 +1,318 @@
+# At the moment it cannot be reused to launch cluster for Origin tests
+kind: Template
+apiVersion: template.openshift.io/v1
+
+parameters:
+- name: JOB_NAME_SAFE
+  required: true
+- name: JOB_NAME_HASH
+  required: true
+- name: LOCAL_IMAGE_BIN
+  required: true
+- name: NAMESPACE
+  required: true
+- name: CLUSTER_TYPE
+  value: "azure"
+  required: true
+- name: AZURE_VM_SIZE
+  value: "Standard_D2s_v3"
+  required: true
+- name: OPENSHIFT_RELEASE
+  value: "3.10"
+  required: true
+- name: IMAGE_ANSIBLE
+  required: true
+
+objects:
+
+# We want the cluster to be able to access these images
+- kind: RoleBinding
+  apiVersion: authorization.openshift.io/v1
+  metadata:
+    name: ${JOB_NAME_SAFE}-image-puller
+    namespace: ${NAMESPACE}
+  roleRef:
+    name: system:image-puller
+  subjects:
+  - kind: SystemGroup
+    name: system:unauthenticated
+
+# The build pod will create an instance and install openshift
+- kind: Pod
+  apiVersion: v1
+  metadata:
+    name: ${JOB_NAME_SAFE}
+    namespace: ${NAMESPACE}
+    annotations:
+      # we want to gather the teardown logs no matter what
+      ci-operator.openshift.io/wait-for-container-artifacts: teardown
+  spec:
+    restartPolicy: Never
+    activeDeadlineSeconds: 10800
+    terminationGracePeriodSeconds: 600
+    volumes:
+    - name: shared-tmp
+      emptyDir: {}
+    - name: shared-data
+      emptyDir: {}
+    - name: cluster-secrets-azure
+      secret:
+        secretName: azure
+
+    containers:
+    - name: buildimage
+      image: ${IMAGE_ANSIBLE}
+      resources:
+        requests:
+          cpu: 1
+          memory: 300Mi
+        limits:
+          cpu: 3
+          memory: 4Gi
+      volumeMounts:
+      - name: shared-tmp
+        mountPath: /tmp/shared
+      - name: cluster-secrets-azure
+        mountPath: /usr/share/ansible/openshift-ansible/inventory/dynamic/injected
+      env:
+      - name: HOME
+        value: /usr/share/ansible/openshift-ansible
+      - name: AZURE_VM_SIZE
+        value: "${AZURE_VM_SIZE}"
+      - name: JOB_NAME_SAFE
+        value: "${JOB_NAME_SAFE}"
+      - name: JOB_NAME_HASH
+        value: "${JOB_NAME_HASH}"
+      - name: OPENSHIFT_RELEASE
+        value: "${OPENSHIFT_RELEASE}"
+      - name: TYPE
+        value: azure
+      - name: INSTANCE_PREFIX
+        value: ${NAMESPACE}-${JOB_NAME_HASH}
+      command:
+      - /usr/local/bin/entrypoint-provider
+      args:
+      - /bin/bash
+      - -c
+      - |
+        #!/bin/bash
+        # error handling and sync code
+        set -euo pipefail
+
+        trap '[ $? == 0 ] || touch /tmp/shared/exit' EXIT
+        trap 'kill $(jobs -p); exit 0' TERM
+
+        # We need to persist this variable so that we can clean it up when we exit
+        echo "RESOURCE_GROUP=${INSTANCE_PREFIX}-$(TZ=Etc/UTC date +%Y%m%d%H%M)" >>/tmp/shared/environment
+        echo "IMAGE_NAME=centos7-$OPENSHIFT_RELEASE-$(TZ=Etc/UTC date +%Y%m%d%H%M)" >>/tmp/shared/environment
```

Comment:
What about RHEL images?

Reply:
Today we have separate jobs for the rhel and centos.  How do you foresee us having a single template and passing in variables for the build job?  I can variable-ize the flavor of OS but how does that get passed into the job?

Reply:
@kargakis, any opinion on how this would work?